src/ng/animate.js

```diff
@@ -416,8 +416,8 @@ var $AnimateProvider = ['$provide', /** @this */ function($provide) {
        * @param {Promise} animationPromise The animation promise that is returned when an animation is started.
        */
       cancel: function(runner) {
-        if (runner.end) {
-          runner.end();
+        if (runner.cancel) {
+          runner.cancel();
```

Comment:
@Narretz, here is an example that shows that when we replace `.then(fn)` with `.done(status => status && fn())`, bad things can happen: [POC][1]. If you remove the `$animate` decorator, everything works.

[1]: https://codepen.io/ExpertSystem/pen/dOoedG?editors=0011